### DISCUSSION TITLE
Running out of space

### DISCUSSION THREAD

**Discussion by jds3d** [mcguirepr89/BirdNET-Pi#716] [General]
I'm going to have to join the group of people who have run out of space.  I have a 64 GB sd card, and my log is currently showing:
![image](https://user-images.githubusercontent.com/6717676/215149518-6ee42e2d-b534-4647-a7e9-a47f92abf762.png)

It looks like all of the space is used up by unprocessed audio:
<img width="332" alt="image" src="https://user-images.githubusercontent.com/6717676/215149600-7e54c945-1da8-4468-9d75-e6ec215e0ac4.png">

I have the Full Disk Behavior set to Purge:
![image](https://user-images.githubusercontent.com/6717676/215149684-71f8a4c8-7e8c-48c8-b44d-d0b59f350a6d.png)

So not sure what I should do here.

**Comment by FirstTenor** [comment]
I saw the same thing earlier today …. And have a very similar setup to as you describe.  Pi 4B+, 64Gb sad card and full disk set to ‚purge‘.   … system was running slowly yesterday and didnt record any new birds after lunch.  It didnt display the full birdnet-pi screen this morning - and now has not come back up following a sudu shutdown -h (and power cycle).

Will need to retry a reboot with a monitor & keyboard next. … but dont understand what has happened

**Comment by Scraggarax** [comment]
This seems like something similar #653. And it's been reported a few times lately.

What I think might be happening, is that when the services are stopped for an update or a sudden loss of power, the analysis service just hangs or needs a restart. And it needs to analyze the recordings before it can purge. You'll have to delete the recordings from birdsongs folder to make space first.

I'm not entirely sure on this part, but sometimes it locks up if I'm not viewing the log as it does it at the same time.

Go into your services, and restart the analysis services. Then go to the log and watch the date and time on the right. You'll see where it's at and just keep an eye on that log until it caches up. If it's been offline for a few days, you're going to have to restart the analysis service a few times to get this working again.

Be sure to do this before your birdnet fills up with recordings. Once it does that, it will not have enough free space to do anything else. You're going to have to go into the file manager and delete the audio files from the birdsong folder to make enough space to do anything.

> I'm going to have to join the group of people who have run out of space. I have a 64 GB sd card, and my log is currently showing: ![image](https://user-images.githubusercontent.com/6717676/215149518-6ee42e2d-b534-4647-a7e9-a47f92abf762.png)
> 
> It looks like all of the space is used up by unprocessed audio: <img alt="image" width="332" src="https://user-images.githubusercontent.com/6717676/215149600-7e54c945-1da8-4468-9d75-e6ec215e0ac4.png">
> 
> I have the Full Disk Behavior set to Purge: ![image](https://user-images.githubusercontent.com/6717676/215149684-71f8a4c8-7e8c-48c8-b44d-d0b59f350a6d.png)
> 
> So not sure what I should do here.

**Comment by jds3d** [comment]
I moved the thursday folder off and restarted all of the services.  It appears to be processing happily now and if it clears enough data, i'll move thursday back on.  Thanks guys!

**Comment by jds3d** [comment]
Is there any way to get it to go back and process all the old files?  It went through Friday but left Monday, Tuesday, and Wednesday (and Thursday is still off the machine.

**Comment by PeterIP4** [comment]
I also have had this problem. It happened the first time and locked verything up.

I then rebuilt the system and for 40 days everything was fine. I applied updates when they were available and monitored storage use which was increasing slowly but hadn't reached 95%.

Then all of a sudden the storage leapt to 100% in a day.

So I need 
a) to know what I can safely delet from where to recover my stoage
b) can I please stop it happening again, any advice will be welcomed.

Thanks all
Peter

**Comment by norfolk-fr** [comment]
Hello everyone,
This strange behavior happened to me a few days ago and it's happening more and more.
For some reason, the analysis service seems to be frozen or does nothing and the /dev/shm file system where the processed data is hosted becomes full (only 512MB allowed on this file system).
After a few days of solving the problem, I found a workaround: when the file system reaches 80%, I stop the recording service so that the data can be processed. When the file system reaches 50%, I start the recording service. In normal operation, the file system does not exceed 30%.
The extracted data is hosted on another file system and has not yet reached 100%.
Here is a little script that does the job in crontab for me until I find something smarter.


```
#!/bin/bash
# FS Path
FILE_SYSTEM="/dev/shm"
# Occupancy %
MAX_OCCUPANCY=80
MIN_OCCUPANCY=50
# Services
RECORDING_SERVICE="birdnet_recording.service"
SERVER_SERVICE="birdnet_server.service"
# Log
LOG_FILE="/home/pi/check_restart.log"
# STATE
ETAT=$(cat "/home/pi/check.txt")
# Check occupancy rate
CURRENT_OCCUPANCY=$(df -h $FILE_SYSTEM | awk 'NR==2 {print $5}' | cut -d'%' -f1)
# Check occupancy rate > MAX_OCCUPANCY
if [ $CURRENT_OCCUPANCY -gt $MAX_OCCUPANCY ] && [ $ETAT == "0" ]; then
  # Stop recording service
  echo "1" > /home/pi/check.txt
  sudo systemctl stop $RECORDING_SERVICE
  # Wait a little bit
  sleep 5
  # Restart the server service (that also restart the analysis service)
  sudo systemctl restart $SERVER_SERVICE
  # Send a Telegram notification
  curl -s -X POST https://api.telegram.org/MYBOT/sendMessage -d chat_id=CHATID -d text="/dev/shm > 80 service recording stoped"
  # Write some logs
  echo "Le taux d'occupation de $FILE_SYSTEM est de $CURRENT_OCCUPANCY%, le service $RECORDING_SERVICE est arrete $(date +'%Y-%m-%d %H:%M:%S')" >> $LOG_FILE
fi
# Occupancy rate < MIN_OCCUPANCY and STATE is 1 (means that recording service is stoped)
if [ $CURRENT_OCCUPANCY -lt $MIN_OCCUPANCY ] && [ $ETAT == "1" ]; then
  echo "0" > /home/pi/check.txt
  # Start recording service
  sudo systemctl start $RECORDING_SERVICE
  # Wait a little bit
  sleep 5 
  # Restart the server service (that also restart the analysis service)
  sudo systemctl restart $SERVER_SERVICE
  # Send a Telegram notification
  curl -s -X POST https://api.telegram.org/MYBOT/sendMessage -d chat_id=CHATID -d text="/dev/shm < 50 service recording started"
  # Write some logs  
  echo "Le taux d'occupation de $FILE_SYSTEM est de $CURRENT_OCCUPANCY%, le service $RECORDING_SERVICE est demarre $(date +'%Y-%m-%d %H:%M:%S')" >> $LOG_FILE
fi
```

Hope this helps

**Comment by oliverk71** [comment]
I have this problem, too, on both of my Pi 3B. Folders aren't moved to the 'Processed' folder and the Pi runs out of free space.

**Reply by oliverk71** [reply]
I feel that it may have something to do with the poor wifi signal.

**Comment by figby** [comment]
I had this issue and discovered it was the logs in /var/log that were filling up my filesystem and not being purged. I found the culprit by using this command: 

find . -maxdepth 1 -mindepth 1 -type d -exec du -sh {} \; | sort -rh | head

Then drilling down into the offending filesystems until I found the issue and deleted the old logs. Hope this helps!

**Reply by lloydbayley** [reply]
Excellent find! We really need a 'Tip & Tricks' place or perhaps something in the wiki for Quick-Fixes.

FYI: You'll probably want to wrap the ; in single quotes to stop the shell trying to interpret it, so it would be:
```find . -maxdepth 1 -mindepth 1 -type d -exec du -sh {} ';' | sort -rh | head```

**Reply by figby** [reply]
I'm going to watch those logs closely (I'm sure they'll fill up again) and see what's going on and will circle back w/what I find. I totally agree we need a "Tips & Tricks" section

**Reply by figby** [reply]
Confirmed that BirdNET-Pi is absolutely crushing my syslog and kernel log and filling up the filesystem. Is there any way to turn down logging to something reasonable? I've created a cronjob that dumps my logs every 6 hours but that's not a great solution.

**Reply by lloydbayley** [reply]
Interesting... My syslog isn't huge.
<img width="597" alt="image" src="https://user-images.githubusercontent.com/21309308/228246386-2b1183e3-7403-4695-9432-5a0e9edaaf6c.png">
Did you specifically enable any debugging anywhere at one stage?

**Comment by figby** [comment]
I didn't AFAIK but it definitely feels like debug-level logging. I'm running it on a RPi Zero 2 with a 64GB microSD.

**Reply by jaredb7** [reply]
Did you happen to have a peek in the syslog file to see if it was anything obvious? I've had the same issue but under a x86 install, both syslog and daemon.log were 500+MB and I just nuked them to recover that space but have no idea what was in those. It did seem to happen out of nowhere too.

There's no log rotation by default in Debian 11 hence just one big file but none the less that's a lot of logging going into that file for some reason.

I have some small fixes lined up to quieten the logs a little, but not sure if t here's stuff happening for you that isn't for me... If this is happening are you able to look in the file and see if anything stands out? you can also tail -f the syslog and see how chatty it is too - things might stand out better.

**Reply by figby** [reply]
Yeah, I took a peek and it's definitely BirdNET that's filling up the logs. I created a small cronjob to zero out the logs every 12 hours, not the most elegant solution  but it's effective.

**Reply by lloydbayley** [reply]
When you say it's BirdNet filling them up, what kind of things are being excessively logged?
Have you a sample (say 10 lines - removing any sensitive data) that you can share here?

**Comment by lloydbayley** [comment]
I know there are places in birdnet to increase debugging levels (in some of the .php files) but you have to go and actually change it to do that. It's really weird. I think I'll have to defer to @ehpersonal38 on this one. He's more 'intimate' with the code at that end. He'll know where to look!

**Comment by gerritzen** [comment]
BirdNET-Pi is also writing a lot in the systems journal. If you don't want to manually delete files, you could first check `journalctl --disk-usage` and do something like `sudo journalctl --vacuum-size=100M`. Maybe the install script should offer (or enforce) a limit of the journal size. If you want to change it yourselves, you can adjust `SystemMaxUse` in `/etc/systemd/journald.conf`.
